code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushNotificationBuilder.java

```diff
@@ -0,0 +1,467 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.TaskStackBuilder;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.Color;
+import android.media.RingtoneManager;
+import android.os.Build;
+import android.widget.RemoteViews;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.core.app.NotificationCompat;
+
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.util.StringUtils;
+
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Class for building push notifications.
+ * <p>
+ * The {@link #buildPushNotification(AEPPushPayload, Context)} method in this class takes the {@link AEPPushPayload} created from the
+ * push notification and builds the notification.
+ * This class is used internally by the {@link AEPMessagingService} to build the push notification.
+ */
+class AEPPushNotificationBuilder {
+
+    private static final String SELF_TAG = "AEPPushNotificationBuilder";
+    private static final String DEFAULT_CHANNEL_ID = "CampaignPushChannel";
+    // When no channel name is received from the push notification, this default channel name is used.
+    // This will appear in the notification settings for the app.
+    private static final String DEFAULT_CHANNEL_NAME = "Campaign Classic General Notifications";
+
+    /**
+     * Builds a notification for the provided {@code AEPPushPayload}.
+     *
+     * @param payload {@link AEPPushPayload} created from the received push notification
+     * @param context the application {@link Context}
+     * @return the notification
+     */
+    @NonNull
+    static Notification buildPushNotification(final AEPPushPayload payload,
+                                              final Context context) {
+        Notification notification;
+        final PushTemplateType pushTemplateType = payload.getPushTemplateType();
+        final AEPPushTemplate pushTemplate = payload.getPushTemplate();
```

Comment:
Using a super type here will result in losing the child specific attributes and inability to pass them to the other utility methods. Recommend passing the payload directly to the `create*TemplatePushNotification` after identifying the type.

```
templateType = payload.getTemplateType();
switch(templateType) {
    case BASIC: 
         createBasicTemplatePushNotification(payload, context);
   case CAROUSEL:
        createCarouselTemplatePushNotification(payload, context);

....
}